corehq/apps/export/export.py

```diff
@@ -128,18 +128,26 @@ def get_export_file(export_instances, filters, progress_tracker=None):
     with writer.open(export_instances):
         for export_instance in export_instances:
             # TODO: Don't get the docs multiple times if you don't have to
-            docs = _get_export_documents(export_instance, filters)
+            docs = _get_export_documents_scroll(export_instance, filters)
             _write_export_instance(writer, export_instance, docs, progress_tracker)
 
     return ExportFile(writer.path, writer.format)
 
 
-def _get_export_documents(export_instance, filters):
+def _get_export_documents_scroll(export_instance, filters):
```

Comment:
You can actually get the results count from the return value of `query.scroll()`. So I don't think a separate `_get_export_documents()` and `_get_export_documents_scroll()` are necessary.

[`scroll()`](https://github.com/dimagi/commcare-hq/blob/09f47685d30b1838685fca819448b4acde784ac0/corehq/apps/es/es_query.py#L189-L198) returns a `ScanResult`, which is a simple object that just implements `__iter__` and stores a count. This allows you to get the results and count in a single request, instead of two. e.g.
```python
docs = query.scroll()
print docs.count
for doc in docs:
   ...
```